### DISCUSSION TITLE
About macaroon tracking is it possible

### DISCUSSION THREAD

**Discussion by GeorgeLamburst** [lightningnetwork/lnd#8409] [Q&A]
I've been unable to figure this out. Thanks for helping me in advance. If I create a few macaroons is there a fingerprint on each node activity showing which macaroon did what? eg if I have three macaroons that have open channel, and a channel is opened, is there a way to know which macaroon was used to do the open? If so, how?

**Comment by guggero** [comment]
That functionality isn't there (yet), at least not out of the box. There is the RPC middleware interceptor (https://lightning.engineering/api-docs/api/lnd/lightning/register-r-p-c-middleware) API that you can run in read-only mode to intercept every incoming request. That would allow you to write a tool that does log every request and the macaroon used for it.
There is [an implementation of that](https://github.com/lightninglabs/lightning-terminal/blob/master/firewall/request_logger.go) in the [Lightning Terminal](https://github.com/lightninglabs/lightning-terminal) repository, but not sure if that does exactly what you're looking for.
